subsys/fb/cfb_shell.c

```diff
@@ -0,0 +1,411 @@
+/** @file
+ * @brief Monochrome Character Framebuffer shell module
+ *
+ * Provide some Character Framebuffer shell commands that can be useful for
+ * testing.
+ */
+
+/*
+ * Copyright (c) 2018 Diego Sueiro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <display/cfb.h>
+
+#define print(_sh, _ft, ...) \
+	shell_fprintf(_sh ? _sh : ctx_shell, SHELL_NORMAL, _ft "\r\n", \
+		      ##__VA_ARGS__)
+#define error(_sh, _ft, ...) \
+	shell_fprintf(_sh ? _sh : ctx_shell, SHELL_ERROR, _ft "\r\n", \
+		      ##__VA_ARGS__)
+
+#define HELP_NONE "[none]"
+#define HELP_INIT "call \"cfb init\" first"
+
+#if defined(CONFIG_SSD1673)
+#define DISPLAY_DRIVER				"SSD1673"
+#endif
+
+#if defined(CONFIG_SSD1306)
+#define DISPLAY_DRIVER				"SSD1306"
+#endif
+
+
+const struct shell *ctx_shell;
```

Comment:
What is the use of this global variable? As far as I can see it is only used in the macros ```print``` and ```error``` but then these macros are always passed a shell pointer.